transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -827,11 +827,14 @@ private void write(Object msg, boolean flush, ChannelPromise promise) {
                 next.invokeWrite(m, promise);
             }
         } else {
+            boolean wakeup = Boolean.TRUE.equals(channel().config().getOption(ChannelOption.WAKEUP_ON_WRITE));
```

Comment:
can you run a benchmark to see the impact caching this somewhere vs calling it every time has when writing from outside the event loop?